lib/spack/spack/schema/config.py

```diff
@@ -20,7 +20,18 @@
         "type": "object",
         "default": {},
         "properties": {
-            "shared_linking": {"type": "string", "enum": ["rpath", "runpath"]},
+            "shared_linking": {
+                "anyOf": [
```

Comment:
looks like you already support the string in the schema; it just isn't handled in the code?

Reply:
it is, `update` is always called after reading config

Reply:
Ok awesome — thanks. I missed that you were already using update.